src/GafferScene/Unencapsulate.cpp

```diff
@@ -51,18 +49,90 @@ using namespace GafferScene;
 namespace
 {
 
-// TODO - is there a centralized version of this?
 ScenePlug::ScenePath concatScenePath( const ScenePlug::ScenePath &a, const ScenePlug::ScenePath &b )
 {
-	ScenePlug::ScenePath result = a;
-	result.reserve( result.size() + b.size() );
-	for( const std::string & n : b )
-	{
-		result.push_back( n );
-	}
+	ScenePlug::ScenePath result;
+	result.reserve( a.size() + b.size() );
+	result.insert( result.end(), a.begin(), a.end() );
+	result.insert( result.end(), b.begin(), b.end() );
 	return result;
 }
 
+class CapsuleScope : boost::noncopyable
+{
+
+	public :
+
+		CapsuleScope(
+			const Gaffer::Context *context, const ScenePlug *inPlug,
+			const ScenePlug::ScenePath &parentPath, boost::optional<const ScenePlug::ScenePath &> branchPath, bool requireCapsule, boost::optional<const InternedString &> setName = boost::optional<const InternedString &>()
+		)
```

Comment:
I'd be tempted to have one constructor for the `branchPath` case and one for the `setName` case, for a bit of extra clarity. They could both delegate to a private constructor which initialised `m_scope/m_capsule`. Not super important since this is a private class though - up to you.

Reply:
Have done the separate constructors.